controllers/gatekeeper_controller.go

```diff
@@ -624,3 +635,86 @@ func setContainerArg(obj *unstructured.Unstructured, containerName, argName stri
 		return unstructured.SetNestedStringSlice(container, args, "args")
 	})
 }
+
+func setCurrentNamespace(obj *unstructured.Unstructured, asset, namespace string) error {
+	if obj.GetNamespace() != "" {
+		obj.SetNamespace(namespace)
+	}
+	if err := setClientConfigNamespace(obj, asset, namespace); err != nil {
+		return err
+	}
+	if err := setControllerManagerExceptNamespace(obj, asset, namespace); err != nil {
+		return err
+	}
+	if err := setRoleBindingSubjectNamespace(obj, asset, namespace); err != nil {
+
+	}
+	return nil
+}
+
+func setClientConfigNamespace(obj *unstructured.Unstructured, asset, namespace string) error {
+	if asset != ValidatingWebhookConfiguration {
+		return nil
+	}
+	webhooks, found, err := unstructured.NestedSlice(obj.Object, "webhooks")
+	if err != nil || !found {
+		return errors.Wrapf(err, "Failed to retrieve webhooks definition")
+	}
+	for _, w := range webhooks {
+		webhook := w.(map[string]interface{})
+		if err := unstructured.SetNestedField(webhook, namespace, "clientConfig", "service", "namespace"); err != nil {
+			return errors.Wrapf(err, "Failed to set webhook clientConfig.service.namespace")
+		}
+	}
+	if err := unstructured.SetNestedSlice(obj.Object, webhooks, "webhooks"); err != nil {
+		return errors.Wrapf(err, "Failed to set webhooks")
+	}
+	return nil
+}
+
+func setControllerManagerExceptNamespace(obj *unstructured.Unstructured, asset, namespace string) error {
+	if asset != WebhookFile {
+		return nil
+	}
+	return setContainerArg(obj, managerContainer, ExemptNamespaceArg, namespace)
+}
+
+func setRoleBindingSubjectNamespace(obj *unstructured.Unstructured, asset, namespace string) error {
+	if asset != ClusterRoleBindingFile && asset != RoleBindingFile {
+		return nil
+	}
+	subjects, found, err := unstructured.NestedSlice(obj.Object, "subjects")
+	if !found || err != nil {
+		return errors.Wrapf(err, "Failed to retrieve subjects from roleBinding")
+	}
+	for _, s := range subjects {
+		subject := s.(map[string]interface{})
+		if err := unstructured.SetNestedField(subject, namespace, "namespace"); err != nil {
+			return errors.Wrapf(err, "Failed to set namespace to rolebinding subject")
+		}
+	}
+	if err := unstructured.SetNestedSlice(obj.Object, subjects, "subjects"); err != nil {
+		return errors.Wrapf(err, "Failed to set updated subjects to rolebinding")
+	}
+	return setContainerArg(obj, managerContainer, ExemptNamespaceArg, namespace)
+}
+
+// toMap Convenience method to convert any struct into a map
+func toMap(obj interface{}) map[string]interface{} {
```

Comment:
These functions below look to be left over from a previous PR.

Reply:
Mistake while rebasing, thanks